File: 1642-water-bottles/water-bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int drank = numBottles, empty = numBottles, extra = empty % numExchange;
        numBottles = empty / numExchange;
        while(empty/numExchange > 0){
            drank += numBottles;
            empty = numBottles + extra;
            extra = empty % numExchange;
            numBottles = empty / numExchange;
        }
        return drank;
    }
}

File: 1642-water-bottles/water-bottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function (numBottles, numExchange) {
    let drunk = numBottles;
    let empty = drunk;
    let extra = 0;

    while (Math.floor(empty / numExchange) > 0) {
        numBottles = Math.floor(empty / numExchange);
        extra = empty % numExchange;
        drunk += numBottles;
        empty = numBottles + extra;
    }

    return drunk;
};